app/app/controllers/concerns/cbv/payments_helper.rb

```diff
@@ -38,7 +38,9 @@ def parse_payments(payments)
     end
   end
 
-  def fetch_end_user_account_ids
-    pinwheel.fetch_accounts(end_user_id: @cbv_flow.pinwheel_end_user_id)["data"].map { |account| account["id"] }
+  def fetch_known_end_user_account_ids
```

Comment:
Would `.pluck` make more sense here since we don't need to fetch the entire column, just the `id`?

```suggestion
def fetch_known_end_user_account_ids
  pinwheel_account_ids = pinwheel.fetch_accounts(end_user_id: @cbv_flow.pinwheel_end_user_id)["data"].pluck("id")
  PinwheelAccount.where(pinwheel_account_id: pinwheel_account_ids).pluck(:pinwheel_account_id)
end
```